crates/biome_js_analyze/src/lint/nursery/use_components_only_module.rs

```diff
@@ -0,0 +1,347 @@
+    /// ```js
+    /// export function hoge() {
+    ///   return 100
+    /// }
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// ### allowConstantExport
+    ///
+    /// Some frameworks, such as Vite, allow exporting constants along with components. By enabling the following, you can avoid warnings.
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options":{
+    ///         "allowConstantExport" : true
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### allowExportNames
+    ///
+    /// If you use a framework that handles HMR of some specific exports, you can use this option to avoid warning for them.
+    ///
+    /// Example for [Remix](https://remix.run/docs/en/main/discussion/hot-module-replacement#supported-exports):
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options":{
+    ///         "allowExportNames": ["json", "loader", "headers", "meta", "links", "scripts"]
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### checkJS
+    ///
+    /// If you are using JSX within .js files, you can apply the following to enable it for .js files as well. However, this is generally not recommended.
+    ///
+    /// ```json
+    /// {
+    ///     "//": "...",
+    ///     "options":{
+    ///         "checkJS" : true
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ## More Example
+    ///
+    /// ### invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// export const foo = () => {};
+    /// export const Bar = () => <></>;
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// export default function () {};
+    /// export default compose()(MainComponent);
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// const Tab = () => {};
+    /// export const tabs = [<Tab />, <Tab />];
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// const App = () => {}
+    /// createRoot(document.getElementById("root")).render(<App />);
+    /// ```
+    ///
+    /// ### valid
+    ///
+    /// ```jsx
+    /// export default function Foo() {
+    ///     return <></>;
+    /// }
+    /// ```
+    ///
+    /// ```jsx
+    /// const foo = () => {};
+    /// export const Bar = () => <></>;
+    /// ```
+    ///
+    /// ```jsx
+    /// import { App } from "./App";
+    /// createRoot(document.getElementById("root")).render(<App />);
+    /// ```
+    ///
+    /// Functions that return standard React components are also permitted.
+    ///
+    /// ```jsx
+    /// import { memo } from 'react';
+    /// const Component = () => <></>
+    /// export default memo(Component);
+    /// ```
+    ///
+    pub UseComponentsOnlyModule {
+        version: "1.8.0",
+        name: "useComponentsOnlyModule",
+        language: "jsx",
+        sources: &[RuleSource::EslintReactRefresh("only-export-components")],
+        source_kind: RuleSourceKind::Inspired,
+        recommended: false,
+    }
+}
+
+fn is_default<T: Default + Eq>(value: &T) -> bool {
```

Comment:
I understand that it should be that way. I wrote this based on [biome_analyze/CONTRIBUTING.md](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#rule-configuration) and use_filenaming_convention.rs as references. Those might need to be revised as well. (For now, I deleted it since it doesn't seem like a crucial item.)